### PR TITLE
GEOS-8935: introduced new coordinates formatting options for WFS GML output formats

### PR DESCRIPTION
**Pull request by mbarto**
Enables configuration of new formatting options introduced with GEOT-6119 for WFS GML output formats

### CODE REVIEW

File: src/main/src/main/java/org/geoserver/catalog/FeatureTypeInfo.java [hunk 1]
```diff
@@ -79,6 +79,24 @@
...
+    boolean getPadWithZeros();
+
+    /**
+     * Sets wether float numbers should be formatted right-padding them with zeros.
+     *
+     * @param padWithZeros
+     */
+    void setPadWithZeros(boolean padWithZeros);
+
+    boolean getForcedDecimal();
```
Comment (aaime) [inline src/main/src/main/java/org/geoserver/catalog/FeatureTypeInfo.java]: Interface method without javadocs, please add

File: src/wfs/src/main/java/org/geoserver/wfs/WFSGetFeatureOutputFormat.java [hunk 2]
```diff
@@ -229,6 +238,38 @@ protected int getNumDecimals(List featureCollections, GeoServer geoServer, Catal
...
+    protected boolean getForcedDecimal(
+            List featureCollections, GeoServer geoServer, Catalog catalog) {
+        boolean forcedDecimal = false;
+        for (int i = 0; i < featureCollections.size(); i++) {
+            Boolean forced =
+                    getFeatureTypeInfoProperty(
+                            catalog,
+                            (FeatureCollection) featureCollections.get(i),
+                            fti -> fti.getForcedDecimal());
+            if (forced != null && forced.booleanValue()) {
```
Comment (aaime) [inline src/wfs/src/main/java/org/geoserver/wfs/WFSGetFeatureOutputFormat.java]: This is a bit weird way to do the comparison, ``Boolean.TRUE.equals(forced)`` would be the idiomatic approach

File: src/wfs/src/main/java/org/geoserver/wfs/WFSGetFeatureOutputFormat.java [hunk 3]
```diff
@@ -240,18 +281,14 @@ protected int getNumDecimals(List featureCollections, GeoServer geoServer, Catal
...
-                if (!featureTypeInfo.getEncodeMeasures()) {
-                    // no measures should be encoded
-                    encodeMeasures = false;
-                }
+            Boolean measures =
+                    getFeatureTypeInfoProperty(
+                            catalog,
+                            (FeatureCollection) featureCollections.get(i),
+                            fti -> fti.getEncodeMeasures());
+            if (measures != null && !measures.booleanValue()) {
```
Comment (aaime) [inline src/wfs/src/main/java/org/geoserver/wfs/WFSGetFeatureOutputFormat.java]: Same as above

File: src/wfs/src/test/java/org/geoserver/wfs/xml/GMLOutputFormatTest.java [hunk 4]
```diff
@@ -60,6 +62,35 @@ public void testGML2() throws Exception {
         assertNull(getFirstElementByTagName(dom, "gml:exterior"));
     }
 
+    @Test
+    public void testGML2CoordinatesFormatting() throws Exception {
+        enableCoordinatesFormatting();
```
Comment (aaime) [inline src/wfs/src/test/java/org/geoserver/wfs/xml/GMLOutputFormatTest.java]: Would be best to restore the config the way it was after the test, you could use a @After method to do that.

**Comment by mbarto** [comment]
@aaime your observations should be addressed now

**Comment by aaime** [comment]
Looks good. The failure on app-veyor is one of the random build failures still left on Windows (btw, I'm looking for devs on windows to help with those).

File: src/main/src/main/java/org/geoserver/catalog/impl/FeatureTypeInfoImpl.java [hunk 5]
```diff
@@ -34,6 +34,8 @@
 
     protected int maxFeatures;
     protected int numDecimals;
+    protected boolean padWithZeros;
+    protected boolean forcedDecimal;
```
Comment (tbarsballe) [inline src/main/src/main/java/org/geoserver/catalog/impl/FeatureTypeInfoImpl.java]: I just noticed this now, but you should probably be including these two new parameters in the equals() and hashCode() implementations

**Comment by mbarto** [comment]
Good catch. I will make a quick-fix PR as soon as possible.
Thanks

**Pull request merged**